Assets/HoloToolkit/UX/Scripts/Lines/LineObjectSwarm.cs

```diff
@@ -0,0 +1,112 @@
+        {
+            Vector3 randomPoint = Vector3.one;
+            randomPoint.x = (float)randomPosition.Next(-RandomValueResolution, RandomValueResolution) / (RandomValueResolution * 2);
+            randomPoint.y = (float)randomPosition.Next(-RandomValueResolution, RandomValueResolution) / (RandomValueResolution * 2);
+            randomPoint.z = (float)randomPosition.Next(-RandomValueResolution, RandomValueResolution) / (RandomValueResolution * 2);
+
+            return Vector3.Scale(randomPoint, SwarmScale);
+        }
+
+        public void Update()
+        {
+            UpdateCollection();
+
+            #if UNITY_EDITOR
+            UnityEditor.EditorUtility.SetDirty(gameObject);
+            #endif
+        }
+
+        public void UpdateCollection()
+        {
+            if (source == null)
+                source = gameObject.GetComponent<LineBase>();
+            if (source == null)
+                return;
+
+            if (prevPoints == null || prevPoints.Length != Objects.Count)
+                prevPoints = new Vector3[Objects.Count];
+
+            randomPosition = new System.Random(Seed);
+            Vector3 linePoint = source.GetPoint(NormalizedDistance);
+            Quaternion lineRotation = source.GetRotation(NormalizedDistance, RotationTypeOverride);
+
+            for (int i = 0; i < Objects.Count; i++)
+            {
+                if (Objects[i] == null)
+                    continue;
+                
+                Vector3 point = source.transform.TransformVector (GetRandomPoint());
+                point.x = (float)(point.x + (noise.Evaluate((point.x + AxisOffset.x) * ScaleMultiplier, Time.unscaledTime * AxisSpeed.x * SpeedMultiplier)) * AxisStrength.x * StrengthMultiplier);
+                point.y = (float)(point.y + (noise.Evaluate((point.y + AxisOffset.y) * ScaleMultiplier, Time.unscaledTime * AxisSpeed.y * SpeedMultiplier)) * AxisStrength.y * StrengthMultiplier);
+                point.z = (float)(point.z + (noise.Evaluate((point.z + AxisOffset.z) * ScaleMultiplier, Time.unscaledTime * AxisSpeed.z * SpeedMultiplier)) * AxisStrength.z * StrengthMultiplier);
+
+                Objects[i].position = point + linePoint;
+                if (SwarmVelocities)
+                {
+                    Vector3 velocity = prevPoints[i] - point;
+                    Objects[i].rotation = Quaternion.Lerp (lineRotation, Quaternion.LookRotation(velocity, Vector3.up), VelocityBlend);
+                }
+                else
+                {
+                    Objects[i].rotation = lineRotation;
+                }
+                Objects[i].Rotate(RotationOffset);
+
+                prevPoints[i] = point;
+            }
+        }
+
+        [SerializeField]
+        private LineBase source;
+
+        private Vector3[] prevPoints;
+        private System.Random randomPosition;
+        private System.Random randomRotation;
+        private FastSimplexNoise noise = new FastSimplexNoise();
```

Comment:
nit: fields should be declared at the top of the class

Reply:
Moved.

Reply:
Not seeing this change yet.

Reply:
This has been committed as well.